src/plugins/home/public/application/components/homepage/homepage.tsx

```diff
@@ -0,0 +1,278 @@
+
+    // TODO: added a toast, but is there a better way to surface this error?
+    // eslint-disable-next-line no-console
+    console.error(error);
+  }, [toastNotifications, error]);
+
+  if (error) {
+    return (
+      <EuiFlexGroup
+        className="home-homepage-body--fill"
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiIcon type="alert" color="danger" size="xxl" />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <FormattedMessage
+            id="home.errorText"
+            defaultMessage="There was an error loading the homepage."
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  if (isHomepageLoading || isWelcomeLoading) {
+    return (
+      <EuiFlexGroup
+        className="home-homepage-body--fill"
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiLoadingSpinner
+            aria-label={i18n.translate('home.loadingSpinner', {
+              defaultMessage: 'Loading homepage',
+            })}
+            size="xl"
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <FormattedMessage id="home.loadingText" defaultMessage="Loading..." />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  if (showWelcome) {
+    return (
+      <Welcome
+        urlBasePath={getBasePath()}
+        onSkip={onSkip}
+        telemetry={telemetry}
+        branding={getBranding()}
+        logos={logos}
+      />
+    );
+  }
+
+  const hero = heroes?.[0];
+
+  return (
+    <>
+      {hero && <HeroSection render={hero.render} />}
+      {sections?.map(({ render, title, description, links }, i) => (
+        <Section key={i} title={title} description={description} links={links} render={render} />
+      ))}
+      <EuiHorizontalRule />
+      <Footer />
+    </>
+  );
+};
+
+export const Homepage = () => {
+  const {
+    application: { getUrlForApp },
+    chrome,
+    injectedMetadata: { getBranding },
+  } = getServices();
+
+  const branding = getBranding();
+  // TODO: is there a better way to check if the title is custom?
+  const title =
+    branding.applicationTitle !== 'OpenSearch Dashboards'
+      ? i18n.translate('home.customBrandingTitle', {
+          defaultMessage: '{title} - Home',
+          values: { title: branding.applicationTitle },
+        })
+      : i18n.translate('home.title', { defaultMessage: 'Home' });
+  const mark = (branding.darkMode && branding.mark?.darkModeUrl) || branding.mark?.defaultUrl;
+
+  useMount(() => {
+    chrome.setBreadcrumbs([
+      {
+        text: i18n.translate('home.breadcrumbs.homeTitle', { defaultMessage: 'Home' }),
+      },
+    ]);
+  });
+
+  const sideItems: React.ReactNode[] = [
+    <EuiButtonEmpty
+      iconType="wrench"
+      href={getUrlForApp('dev_tools', { path: '#/console' })}
```

Comment:
Same here, use navigateToApp

Reply:
@ashwin-pc, to expedite this, can you please make the suggestion and then accept it?